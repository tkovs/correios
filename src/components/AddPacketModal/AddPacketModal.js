/* eslint-disable react-native/no-inline-styles */
import React, { useState, useEffect } from 'react'
import PropTypes from 'prop-types'
import { TextInput, HelperText } from 'react-native-paper'
import { View, StyleSheet } from 'react-native'

import Modal from '../Modal'
import { colors } from '../../styles/theme'

function AddPacketModal({ error, pending, visible, onDismiss, onSubmit }) {
  const [title, setTitle] = useState('')
  const [code, setCode] = useState('')

  useEffect(() => {
    if (!pending && !error) {
      onDismiss()
    }
  }, [pending])

  return (
    <Modal
      title="Adicionar novo rastreio"
      visible={visible}
      onSubmit={() => onSubmit(title, code)}
      onDismiss={onDismiss}
      loading={pending}
    >
      <View style={styles.textInputContainer}>
        <TextInput
          theme={{
            colors: { primary: colors.blue, underlineColor: 'transparent' },
          }}
          dense
          placeholder="iPad"
          label="Nome do pacote"
          mode="outlined"
          value={title}
          onChangeText={value => setTitle(value)}
        />
      </View>
      <View>
        <TextInput
          theme={{
            colors: { primary: colors.blue, underlineColor: 'transparent' },
          }}
          autoCapitalize="characters"
          dense
          placeholder="PW086958101BR"
          label="CÃ³digo de rastreio"
          mode="outlined"
          value={code}
          onChangeText={value => setCode(value)}
        />
        <HelperText type="error" visible={!!error}>
          {error}
        </HelperText>
      </View>
    </Modal>
  )
}

AddPacketModal.defaultProps = {
  visible: false,
  error: null,
}

AddPacketModal.propTypes = {
  onDismiss: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  pending: PropTypes.bool.isRequired,
  visible: PropTypes.bool,
  error: PropTypes.string,
}

const styles = StyleSheet.create({})

export default AddPacketModal
