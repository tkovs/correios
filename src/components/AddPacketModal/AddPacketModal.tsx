import React, { FC, useState, useEffect } from 'react'
import { TextInput, HelperText } from 'react-native-paper'
import { View, Keyboard } from 'react-native'
import find from 'lodash/find'

import Modal from '../Modal'
import { colors } from '../../styles/theme'
import { PacketsActionTypes } from '../../store/actions/packets/types'

interface AddPacketModalProps {
  onDismiss: Function
  onSubmit: (title: string, code: string) => PacketsActionTypes
  visible: boolean
  statusList: StatusListItem[]
}

type Props = AddPacketModalProps

const AddPacketModal: FC<Props> = ({
  onDismiss,
  onSubmit,
  statusList,
  visible,
}: Props) => {
  const [title, setTitle] = useState('')
  const [code, setCode] = useState('')
  const status = find(statusList, { code })

  useEffect(() => {
    if (!status?.pending && !status?.error) {
      setTitle('')
      setCode('')
      onDismiss()
    }
  }, [onDismiss, status])

  return (
    <Modal
      title="Adicionar novo rastreio"
      visible={visible}
      onSubmit={(): void => {
        Keyboard.dismiss()
        onSubmit(title, code)
      }}
      onDismiss={(): void => {
        setTitle('')
        setCode('')
        onDismiss()
      }}
      loading={status?.pending}
      disabled={title === '' || code === '' || status?.pending}
    >
      <View>
        <TextInput
          theme={{
            colors: { primary: colors.blue },
          }}
          dense
          placeholder="iPad"
          label="Nome do pacote"
          mode="outlined"
          value={title}
          onChangeText={(value: string): void => {
            setTitle(value)
          }}
          testID="title-text-input"
          accessibilityStates={['disabled']}
        />
      </View>
      <View>
        <TextInput
          theme={{
            colors: { primary: colors.blue },
          }}
          autoCapitalize="characters"
          dense
          placeholder="PW086958101BR"
          label="CÃ³digo de rastreio"
          mode="outlined"
          value={code}
          onChangeText={(value): void => {
            setCode(value)
          }}
          testID="code-text-input"
          accessibilityStates={['disabled']}
        />
        <HelperText type="error" visible={!!status?.error}>
          {status?.error}
        </HelperText>
      </View>
    </Modal>
  )
}

AddPacketModal.defaultProps = {
  visible: false,
  statusList: [],
}

export default AddPacketModal
