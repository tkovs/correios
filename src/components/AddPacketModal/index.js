import { connect } from 'react-redux'

import { addPacket } from '../../store/actions/packets'
import AddPacketModal from './AddPacketModal'

const mapDispatchToProps = dispatch => ({
  onSubmit: (title, code) => dispatch(addPacket(title, code)),
})

const mapStateToProps = ({ packets }) => ({
  pending: packets.pending,
  error: packets.error,
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AddPacketModal)
