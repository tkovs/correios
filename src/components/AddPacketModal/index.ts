import { connect, MapDispatchToProps, MapStateToProps } from 'react-redux'
import { PacketsActionTypes } from '../../store/actions/packets/types'

import { addPacket } from '../../store/actions/packets'
import { statusListSelector } from '../../store/selectors/packets'
import AddPacketModal from './AddPacketModal'

interface StateProps {
  statusList: StatusListItem[]
}

interface OwnProps {
  visible: boolean
  onDismiss: Function
}

interface DispatchProps {
  onSubmit: (title: string, code: string) => PacketsActionTypes
}

const mapDispatchToProps: MapDispatchToProps<DispatchProps, OwnProps> = (
  dispatch
) => ({
  onSubmit: (title, code): PacketsActionTypes =>
    dispatch(addPacket(title, code)),
})

const mapStateToProps: MapStateToProps<StateProps, OwnProps, State> = (
  state: State
) => ({
  statusList: statusListSelector(state),
})

export default connect(mapStateToProps, mapDispatchToProps)(AddPacketModal)
