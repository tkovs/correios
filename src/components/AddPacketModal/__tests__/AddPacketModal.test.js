import 'react-native'
import React from 'react'
import { render } from '@testing-library/react-native'

import AddPacketModal from '../AddPacketModal'

const mockOnDismiss = jest.fn()

describe('AddPacketModal', () => {
  it('renders correctly', () => {
    const { baseElement } = render(<AddPacketModal onDismiss={mockOnDismiss} />)
    expect(baseElement).toMatchSnapshot()
  })
})
