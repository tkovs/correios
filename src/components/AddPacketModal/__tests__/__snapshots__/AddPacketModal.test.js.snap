// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPacketModal renders correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "bottom": "100%",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "paddingBottom": 4,
          "paddingEnd": 4,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <TouchableWithoutFeedback
        hitSlop={
          Object {
            "bottom": 4,
            "left": 4,
            "right": 4,
            "top": 4,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(250, 186, 48, 0.95)",
              },
              Object {
                "borderRadius": 14,
                "height": 28,
                "justifyContent": "center",
                "paddingHorizontal": 12,
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "rgba(255, 255, 255, 1)",
                "fontSize": 12,
                "includeFontPadding": false,
                "lineHeight": 16,
              }
            }
          >
            Dismiss All
          </Text>
        </View>
      </TouchableWithoutFeedback>
    </View>
    <ScrollView
      data={
        Array [
          Object {
            "category": "Warning: Failed prop type: The prop \`pending\` is marked as required in \`AddPacketModal\`, but its value is \`undefined\`.
    in AddPacketModal",
            "warnings": Array [
              YellowBoxWarning {
                "message": Object {
                  "content": "Warning: Failed prop type: The prop \`pending\` is marked as required in \`AddPacketModal\`, but its value is \`undefined\`.
    in AddPacketModal",
                  "substitutions": Array [],
                },
                "stack": Array [
                  Object {
                    "arguments": Array [],
                    "column": 15,
                    "file": "/Users/tkovs/Code/tracking/node_modules/prop-types/checkPropTypes.js",
                    "lineNumber": 20,
                    "methodName": "printWarning",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 11,
                    "file": "/Users/tkovs/Code/tracking/node_modules/prop-types/checkPropTypes.js",
                    "lineNumber": 82,
                    "methodName": "checkPropTypes",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 5,
                    "file": "/Users/tkovs/Code/tracking/node_modules/react/cjs/react.development.js",
                    "lineNumber": 1796,
                    "methodName": "validatePropTypes",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 5,
                    "file": "/Users/tkovs/Code/tracking/node_modules/react/cjs/react.development.js",
                    "lineNumber": 1969,
                    "methodName": "Object.createElementWithValidation [as createElement]",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 36,
                    "file": "/Users/tkovs/Code/tracking/src/components/AddPacketModal/__tests__/AddPacketModal.test.js",
                    "lineNumber": 11,
                    "methodName": "Object.<anonymous>",
                  },
                ],
                "symbolicated": Object {
                  "error": null,
                  "stack": null,
                  "status": "NONE",
                },
              },
            ],
          },
          Object {
            "category": "Warning: Failed prop type: The prop \`onSubmit\` is marked as required in \`AddPacketModal\`, but its value is \`undefined\`.
    in AddPacketModal",
            "warnings": Array [
              YellowBoxWarning {
                "message": Object {
                  "content": "Warning: Failed prop type: The prop \`onSubmit\` is marked as required in \`AddPacketModal\`, but its value is \`undefined\`.
    in AddPacketModal",
                  "substitutions": Array [],
                },
                "stack": Array [
                  Object {
                    "arguments": Array [],
                    "column": 15,
                    "file": "/Users/tkovs/Code/tracking/node_modules/prop-types/checkPropTypes.js",
                    "lineNumber": 20,
                    "methodName": "printWarning",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 11,
                    "file": "/Users/tkovs/Code/tracking/node_modules/prop-types/checkPropTypes.js",
                    "lineNumber": 82,
                    "methodName": "checkPropTypes",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 5,
                    "file": "/Users/tkovs/Code/tracking/node_modules/react/cjs/react.development.js",
                    "lineNumber": 1796,
                    "methodName": "validatePropTypes",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 5,
                    "file": "/Users/tkovs/Code/tracking/node_modules/react/cjs/react.development.js",
                    "lineNumber": 1969,
                    "methodName": "Object.createElementWithValidation [as createElement]",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 36,
                    "file": "/Users/tkovs/Code/tracking/src/components/AddPacketModal/__tests__/AddPacketModal.test.js",
                    "lineNumber": 11,
                    "methodName": "Object.<anonymous>",
                  },
                ],
                "symbolicated": Object {
                  "error": null,
                  "stack": null,
                  "status": "NONE",
                },
              },
            ],
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      scrollEnabled={false}
      scrollEventThrottle={50}
      scrollsToTop={false}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "height": 97,
        }
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View
        style={null}
      >
        <TouchableWithoutFeedback>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(250, 186, 48, 0.95)",
                },
                Object {
                  "height": 48,
                  "justifyContent": "center",
                  "marginTop": 0.5,
                  "paddingHorizontal": 12,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                numberOfLines={2}
                style={
                  Object {
                    "color": "rgba(255, 255, 255, 1)",
                    "flex": 1,
                    "fontSize": 14,
                    "includeFontPadding": false,
                    "lineHeight": 18,
                  }
                }
              >
                <Text>
                  Warning: Failed prop type: The prop \`pending\` is marked as required in \`AddPacketModal\`, but its value is \`undefined\`.
    in AddPacketModal
                </Text>
              </Text>
            </View>
          </View>
        </TouchableWithoutFeedback>
      </View>
      <View
        style={null}
      >
        <TouchableWithoutFeedback>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(250, 186, 48, 0.95)",
                },
                Object {
                  "height": 48,
                  "justifyContent": "center",
                  "marginTop": 0.5,
                  "paddingHorizontal": 12,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                numberOfLines={2}
                style={
                  Object {
                    "color": "rgba(255, 255, 255, 1)",
                    "flex": 1,
                    "fontSize": 14,
                    "includeFontPadding": false,
                    "lineHeight": 18,
                  }
                }
              >
                <Text>
                  Warning: Failed prop type: The prop \`onSubmit\` is marked as required in \`AddPacketModal\`, but its value is \`undefined\`.
    in AddPacketModal
                </Text>
              </Text>
            </View>
          </View>
        </TouchableWithoutFeedback>
      </View>
    </ScrollView>
    <SafeAreaView
      style={
        Object {
          "backgroundColor": "rgba(250, 186, 48, 0.95)",
          "marginTop": 0.5,
        }
      }
    />
  </View>
</View>
`;
